From e0757b737833c0baee39cf54de96abb15a7b4a24 Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsht@ti.com>
Date: Fri, 21 Jul 2023 19:43:19 +0530
Subject: [PATCH] drm/tidss: Make zpos immutable for primary plane

The new version of weston seems to be setting
non-zero value of zpos for primary plane too [1].

As primary plane sits at the bottom and is always
"on" and overlay planes are "overlayed" on top
of primary, set zpos value of primary plane as
immutable with 0 value.

For overlay plane, set minimum value in range for zpos
value as 1 so that it doesn't overlap with
value for primary plane.

This is based off :
https://lore.kernel.org/linux-renesas-soc/20200401061100.7379-1-etom@igel.co.jp/
https://patchwork.kernel.org/project/linux-renesas-soc/patch/20200402104035.13497-1-laurent.pinchart+renesas@ideasonboard.com/

[1]: http://serenity.dal.design.ti.com:7777/ayoviboceg.yaml

Signed-off-by: Devarsh Thakkar <devarsht@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/gpu/drm/tidss/tidss_plane.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/tidss/tidss_plane.c b/drivers/gpu/drm/tidss/tidss_plane.c
index 5328d8b8cff0..f3e3de71f69b 100644
--- a/drivers/gpu/drm/tidss/tidss_plane.c
+++ b/drivers/gpu/drm/tidss/tidss_plane.c
@@ -211,8 +211,11 @@ struct tidss_plane *tidss_plane_create(struct tidss_device *tidss,
 
 	drm_plane_helper_add(&tplane->plane, &tidss_plane_helper_funcs);
 
-	drm_plane_create_zpos_property(&tplane->plane, hw_plane_id, 0,
-				       num_planes - 1);
+	if (type == DRM_PLANE_TYPE_PRIMARY)
+		drm_plane_create_zpos_immutable_property(&tplane->plane, 0);
+	else
+		drm_plane_create_zpos_property(&tplane->plane, hw_plane_id, 1,
+					       num_planes - 1);
 
 	ret = drm_plane_create_color_properties(&tplane->plane,
 						color_encodings,
-- 
2.17.1

